dsl_parser/tests/test_plugin_properties.py

```diff
@@ -0,0 +1,185 @@
+import pytest
+from mock import patch
+
+from dsl_parser.exceptions import (DSLParsingFormatException,
+                                   DSLParsingLogicException)
+from dsl_parser.tests.abstract_test_parser import AbstractTestParser
+
+PLUGINS_YAML = """
+plugins:
+  dummy:
+    executor: central_deployment_agent
+    package_name: testing
+    source: dummy
+    properties_description: |
+      Description regarding the credentials and
+      the link to AWS documentation how to generate
+      or the link to cloudify documentation on
+      different types of authentication methods
+    properties:
+      aws_access_key_id:
+        type: string
+        description: This is a AWS Access Key ID
+        display_label: AWS Access Key ID
+      aws_secret_access_key:
+        type: string
+        description: This is a AWS Secret Access Key
+        display_label: AWS Secret Access Key
+      aws_region:
+        type: string
+        display_label: AWS Region
+"""
+
+NODES_YAML = """
+node_types:
+  type: {}
+
+node_templates:
+  dummy_node:
+    type: type
+    interfaces:
+      dummy_install:
+        install: dummy.install
+"""
+
+
+class PluginPropertiesTest(AbstractTestParser):
+    def test_parse_1_4(self):
+        with pytest.raises(DSLParsingLogicException,
+                           match='not supported in version cloudify_dsl_1_4'):
+            self.parse_1_4(PLUGINS_YAML + NODES_YAML)
+
+    def test_parse_1_5(self):
+        plan = self.parse_1_5(PLUGINS_YAML + NODES_YAML)
+        dummy = plan['deployment_plugins_to_install'][0]
+        assert dummy['properties_description'].rstrip()\
+            .endswith('authentication methods')
+        assert all(p['type'] == 'string' for p in dummy['properties'].values())
+
+    def test_no_properties(self):
+        yaml = """
+plugins:
+  dummy:
+    executor: central_deployment_agent
+    package_name: testing
+    source: dummy
+    properties_description: Empty credentials
+
+node_types:
+  type: {}
+
+node_templates:
+  dummy_node:
+    type: type
+    interfaces:
+      dummy_install:
+        install: dummy.install
+"""
+        plan = self.parse_1_5(yaml)
+        assert plan['deployment_plugins_to_install'][0]['properties'] == {}
```

Comment:
Should this really be an empty dict? I would've expected it should either be an error (if we require the properties) or a dict containing all the possible properties with them set to None?

Reply:
Yes, the plugin hasn't got any properties, so an empty dict should be appropriate to represent that

Reply:
Doesn't the plugin have three aws_<thing> properties defined?

Reply:
No, it does not, we're testing this yaml in this case:
```yaml
plugins:
  dummy:
    executor: central_deployment_agent
    package_name: testing
    source: dummy
    properties_description: Empty credentials

node_types:
  type: {}

node_templates:
  dummy_node:
    type: type
    interfaces:
      dummy_install:
        install: dummy.install
```